### PR TITLE
fix issues related to DialogBusy

### PR DESCRIPTION
**Pull request by FernetMenta**
- fixes multiple updates of GUIMediaWindow's vecItems at the same time by ignoring updates/refreshes, if update is already in progress

- do not use background loading when context menu (GUiMediaWindow) is active. This resulted in an busyDialog obscured by a selectDialog

- do not bring up busy dialog for GUIMediaWindow if it is in use

- python scripts must not activate busyDialog. 

fixes .i.e. https://trac.kodi.tv/ticket/17896#comment:2
also (most likely) https://trac.kodi.tv/ticket/17892

### CODE REVIEW

File: xbmc/interfaces/legacy/Dialog.cpp [hunk 1]
```diff
@@ -499,59 +500,26 @@ namespace XBMCAddon
...
-      CGUIDialogBusy* pDialog = CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogBusy>(WINDOW_DIALOG_BUSY);
-
-      if (pDialog == nullptr)
-        throw WindowException("Error: Window is NULL, this is not possible :-)");
-
-      dlg = pDialog;
-      open = true;
-
-      pDialog->Open();
+      CLog::Log(LOGWARNING, "using DialogBusy from python deprecated and results in nop");
```
[the remark this answers is not in the document]
Comment (FernetMenta) [reply xbmc/interfaces/legacy/Dialog.cpp]: I updated the comment to: "using DialogBusy from python results in nop now"

Currently I am fixing crashes caused by mistakes in the past. No time for adding an alternative at this stage. I also updated the doxy. If you think I forgot something please PR. I have to move on the the next bug.

**Comment by DaveTBlake** [comment]
Just to link up the related PRs - this makes `xbmcgui.DialogBusy()` added by #10699 simply produce a log entry. But a workaround is provided by #13958

**Comment by candrews** [comment]
I just tested a build of kodi 663fbe4 + this PR + #13958 and found that https://trac.kodi.tv/ticket/17892 is now fixed.

**Comment by DaveTBlake** [comment]
From this commit on music playback in "party mode" is broken

**Pull request merged**